### PR TITLE
Add false to conditional-rendering

### PR DESCRIPTION
**Pull request by donavon**
Conditional rendering can return `null` or `false`. Add the `false` condition and show an example that benefits from using `false` over `null`.

Here's the current example:
```js
function WarningBanner(props) {
  if (!props.warn) {
    return null;
  }

  return (
    <div className="warning">
      Warning!
    </div>
  );
}
```

But by returning `false`, you can do this:
```javascript
const WarningBanner = ({ warn }) => (
  warn && (
    <div className="warning">
      Warning!
    </div>
  )
);
```

### CODE REVIEW

**Comment by donavon** [comment]
FYI... Direct link to changed docs:

https://deploy-preview-1103--reactjs.netlify.com/docs/conditional-rendering.html#preventing-component-from-rendering

File: content/docs/conditional-rendering.md [hunk 1]
```diff
@@ -238,6 +238,18 @@ ReactDOM.render(
 );
 ```
 
+In this refactored version of `<WarningBanner />` (below), you can see the benefits of returning `false` over `null`. Here we use a logical AND to return either `false` or the rendered warning.
```
Comment (alexkrolick) [inline content/docs/conditional-rendering.md]: Would rather explain the benefit than leave it as an exercise to the reader. I assume you are saying the `&&` is shorter and inline-able?

Comment (donavon) [reply content/docs/conditional-rendering.md]: Any better?
> In this refactored version of `<WarningBanner />` (below), the benefits of returning `false` over `null` is that `false` is the natural result of `props.warn` – i.e. no additional values need to be returned after the condition is determined. Here we are using a logical AND to return either `false` or the rendered warning, or to put it in plain English: "return `false` if the `warn` prop is `false`, otherwise render the warning".

I'd also like to link to [this article](https://medium.freecodecamp.org/conditional-rendering-in-react-using-ternaries-and-logical-and-7807f53b6935) to explain it more, but not sure about Facebook's policy on this.

**Comment by BartoszKlonowski** [comment]
Hello @donavon, thank you for the contribution!
From what I can see the example with `WarningBanner` is already removed from the docs, also the conditional rendering now is described with it's own separate section, in my opinion efficiently enough:
https://github.com/reactjs/react.dev/blob/e33ec30515561d715e35b2c437402c5c58d4ecd8/src/content/learn/conditional-rendering.md?plain=1#L263
Please, let me close this PR as already outdated, but we welcome more contributions on other still open issues.
